src/core/sdk.ts

```diff
@@ -133,16 +134,22 @@ export class ThirdwebSDK extends RPCConnectionHandler {
    */
   public wallet: UserWallet;
 
+  /**
+   * Upload and download files from IPFS
+   */
+  public storage: Storage;
+
   constructor(
     network: ChainOrRpc | SignerOrProvider,
     options: SDKOptions = {},
     storage: IStorage = new IpfsStorage(),
   ) {
     const rpc = getProviderForNetwork(network);
     super(rpc, options);
-    this.storage = storage;
-    this.deployer = new ContractDeployer(rpc, options, storage);
-    this.wallet = new UserWallet(rpc, options);
+    this.storageHandler = storage;
+    this.storage = new Storage(storage);
+    this.deployer = new ContractDeployer(network, options, storage);
+    this.wallet = new UserWallet(network, options);
```

Comment:
@adam-maj careful here - you're passing`network` instead of the transformed provider `rpc` (should prob be renamed)

this will def break things

Reply:
Hmm good catch - typing doesn't throw here understandably (maybe because of overlapping types?)